### PR TITLE
refs #61 - update to polymer 2 - classes

### PR DESCRIPTION
**Pull request by Wurper**
* Updated to classes
* Updated fire, async etc
* Tests currently failing

### CODE REVIEW

File: cosmoz-tab.js [hunk 1]
```diff
@@ -52,84 +54,85 @@
...
 			'_notifyProperty("disabled", disabled)',
 			'_notifyProperty("heading", heading)',
 			'_notifyProperty("badge", badge)',
 			'_onAccordionChangedRender(accordion)'
-		],
+		];
+	}
 
-		listeners: {
+	static get listeners() {
```
Comment (megheaiulian) [inline cosmoz-tab.js]: This is not going to work in Polymer 2. Events need to be handled separately.

File: cosmoz-tabs.js [hunk 2]
```diff
@@ -45,75 +45,76 @@
...
-		},
-
-		behaviors: [
-			Cosmoz.TabbableBehavior
-		],
+		};
+	}
 
-		listeners: {
+	static get listeners() {
```
Comment (megheaiulian) [inline cosmoz-tabs.js]: No listeners in 2.x

**Comment by Wurper** [comment]
Need help with this one.
the tests in `test/tabbed-template-behavior.html` fails. 
The inner `<template-bind-test-helper>` does not seem to render.

@megheaiulian Something you can have a look at?

**Comment by megheaiulian** [comment]
@Wurper I'll have a look and update with fixes.

File: cosmoz-tabbable-behavior.html [hunk 3]
```diff
@@ -171,19 +192,27 @@
...
-				this.debounce('select-resize', this.notifyResize);
+				this._debouncer = Polymer.Debouncer.debounce(this._debouncer,
+					Polymer.Async.microTask,
+					this.notifyResize
+				);
+				Polymer.enqueueDebouncer(this._debouncer);
 
 				if (!item.isActive) {
-					this.fire('tab-first-select', {}, opts);
+					item.dispatchEvent(new CustomEvent('tab-first-select', opts));
```
Comment (megheaiulian) [inline cosmoz-tabbable-behavior.html]: @Wurper This was the reason why the tests were failing. The event must be dispatched on the item that gets selected.

Comment (Wurper) [reply cosmoz-tabbable-behavior.html]: Ah! Thanks!

**Pull request merged**